src/components/DatePicker/DatePicker.jsx

```diff
@@ -97,52 +112,59 @@ export const DatePicker = ({
         onMonthChange?.(m);
     };
 
+    const handleStartMonthChange = (m) => {
+        setStartMonth(m);
+        onMonthChange?.(m);
+    };
+
+    const handleEndMonthChange = (m) => {
+        setEndMonth(m);
+        onMonthChange?.(m);
+    };
+
     const captionElement = shouldShowYearPicker
         ? ({ date }) => <MonthYearSelector date={date} currentMonth={currentMonth} onChange={handleMonthChange} />
         : undefined;
 
-    const isDisabled = (date) => {
+    const captionStartElement = shouldShowYearPicker
+        ? ({ date }) => <MonthYearSelector date={date} currentMonth={currentMonth} onChange={handleStartMonthChange} />
+        : undefined;
+
+    const captionEndElement = shouldShowYearPicker
+        ? ({ date }) => <MonthYearSelector date={date} currentMonth={currentMonth} onChange={handleEndMonthChange} />
+        : undefined;
```

Comment:
This was incorrectly named before. Since this returns a React node, follow the naming convention and name it in proper casing like `CaptionStartElement`. Infact you can extract it outside of this component and make it generic.

```jsx
const CaptionElement = ({ date, currentMonth, onChange }) => {
  return whatever_you_need;
});
```

Then you can use the same component with just different props.